### PR TITLE
Feat/agentdatadog provide a test agent to try the datadog agent startup

### PR DESCRIPTION
**Pull request by smerle33**
provide an agent with label test-datadog that should have a datadog-agent configured and launched at startup ....

### CODE REVIEW

File: hieradata/vagrant/common.yaml [hunk 1]
```diff
@@ -69,7 +69,7 @@ profile::jenkinscontroller::jcasc:
         credentialsId: "ppc64le-ed25519-2021-12-03"
         launchTimeoutSeconds: 713
         retryWaitTime: 175
-        hostKey: "ssh-ed25519 whatever"
+        hostKey: "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCYwX3Fu4fWEb9hPlan3FsiUsZoCsMD7CMqFNT/+Uh11HhvKU5ZFiF3sGNI7FrhNNbqBFd0HnS1t3zIkP3FFlKToSTrkcXPUyw+svFf5YbPbDxQUNE0kclTUsERzC3GNB5eXUlyNxCiGksqGtinXgknF2Z5cOO8osODP7ddRc6L3H4gvDi0/smz9QukZB2N0FqBJ3EZGbv0X9V3iwRu6Cu9lhcl/ue5fuIjKAgGzGQgWgCEg0k9xkK0OmxyJalI0eiqBRbdES/bXkkxp+4TQNOsxN/ZrZqxtlN7o9Fq9y+dp7xQFEoivSAjn6WQ6izjkMGKon7rcFJ4t4g6FJoQYv2P"
```
Comment (dduportal) [inline hieradata/vagrant/common.yaml:72]: Side note for later reader: this line change fix the vagrant provisioning (an invalid RSA key fails the provisioning)

**Review by dduportal — APPROVED** [review]

File: hieradata/clients/azure.ci.jenkins.io.yaml [hunk 2]
```diff
@@ -284,6 +284,35 @@ profile::jenkinscontroller::jcasc:
...
+          maxInstances: 1
+          useAsMuchAsPosible: false
+          credentialsId: "jenkinsvmagents-userpass"
+          usePrivateIP: true
+          virtualNetworkName: "prod-jenkins-public-prod"
+          virtualNetworkResourceGroupName: "prod-jenkins-public-prod"
+          subnetName: "ci.j-agents-vm"
+          spot: true
+          executeInitScriptAsRoot: true
+          initScript:
```
Comment (timja) [inline hieradata/clients/azure.ci.jenkins.io.yaml:306]: initScript is a string not an array

see example:
https://github.com/hmcts/cnp-flux-config/blob/866ffa780479849229dafc4fdeae39d0a5de2214/apps/jenkins/jenkins/ptl-intsvc/jenkins.yaml#L543-L565

Comment (dduportal) [reply hieradata/clients/azure.ci.jenkins.io.yaml:306]: Yes, in jcasc :) here you have hieradata. Alas the yaml ruby parser does not manage efficiently the multiline strings so we used array of string for this . Template then iterates on the array to make the multiline block in jcasc final

Comment (timja) [reply hieradata/clients/azure.ci.jenkins.io.yaml:306]: I see odd

Comment (dduportal) [reply hieradata/clients/azure.ci.jenkins.io.yaml:306]: No problem, thanks for post-checking: it is healthy and ensure that we do not miss anything!

**Pull request merged**